moped-editor/src/views/projects/projectView/SubprojectFundingModal.js

```diff
@@ -0,0 +1,169 @@
+import React, { useState } from "react";
+import {
+  Button,
+  Box,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  IconButton,
+  Typography,
+} from "@material-ui/core";
+import { makeStyles } from "@material-ui/core";
+import CloseIcon from "@material-ui/icons/Close";
+import AddCircle from "@material-ui/icons/AddCircle";
+import { useSocrataJson } from "src/utils/socrataHelpers";
+import MaterialTable from "@material-table/core";
+
+import typography from "../../../theme/typography";
+
+const useStyles = makeStyles((theme) => ({
+  dialogTitle: {
+    color: theme.palette.primary.main,
+    fontFamily: theme.typography.fontFamily,
+    display: "flex",
+    justifyContent: "space-between",
+    alignItems: "center",
+  },
+}));
+
+const SubprojectFundingModal = ({
+  isDialogOpen,
+  handleDialogClose,
+  eCaprisID,
+  fdusArray,
+  addProjectFunding,
+  userId,
+  projectId,
+  setSnackbarState,
+}) => {
+  const classes = useStyles();
+  const typographyStyle = {
+    fontFamily: typography.fontFamily,
+    fontSize: "14px",
+  };
+
+  const { data } = useSocrataJson(
+    `https://data.austintexas.gov/resource/jega-nqf6.json?dept_unit_status=Active&sp_number_txt=${eCaprisID}&$limit=9999`
+  );
```

Comment:
This could be a dataset issue and not a code issue but I noticed for all the example eCapris IDs you gave, each of the funding sources has duplicates. Should we dedupe the responses from Socrata? Maybe using the `distinct` keyword in the query? Or revisit the data in dataset?

![Screen Shot 2022-08-25 at 2 34 57 AM](https://user-images.githubusercontent.com/5697474/186603910-e8972249-28e8-4808-9d75-8b378622f7d4.png)

Reply:
you're right its a dataset issue -- the upsert is just adding new rows instead of updating rows. I'm working on fixing it. So glad you caught it.

Reply:
the pesky Socrata row ID strikes again

Reply:
@mateoclarke this gif is some of your finest work

Reply:
this gif is the best